### PR TITLE
gh-96959: Update HTTP links which are redirected to HTTPS

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-96959

### CODE REVIEW

**Review by ronaldoussoren — APPROVED** [review]
LGTM

File: Doc/faq/library.rst [hunk 1]
```diff
@@ -181,7 +181,7 @@ How do I create documentation from doc strings?
 The :mod:`pydoc` module can create HTML from the doc strings in your Python
 source code.  An alternative for creating API documentation purely from
 docstrings is `epydoc <http://epydoc.sourceforge.net/>`_.  `Sphinx
```
Comment (ezio-melotti) [inline Doc/faq/library.rst:183]: This only has `http` -- https://epydoc.sourceforge.net/ fails.

File: LICENSE [hunk 2]
```diff
@@ -2,7 +2,7 @@ A. HISTORY OF THE SOFTWARE
 ==========================
 
 Python was created in the early 1990s by Guido van Rossum at Stichting
-Mathematisch Centrum (CWI, see http://www.cwi.nl) in the Netherlands
+Mathematisch Centrum (CWI, see https://www.cwi.nl) in the Netherlands
```
Comment (serhiy-storchaka) [inline LICENSE]: It is already https in Doc/license.rst.

File: LICENSE [hunk 3]
```diff
@@ -19,7 +19,7 @@ https://www.python.org/psf/) was formed, a non-profit organization
 created specifically to own Python-related Intellectual Property.
 Zope Corporation was a sponsoring member of the PSF.
 
-All Python releases are Open Source (see http://www.opensource.org for
+All Python releases are Open Source (see https://opensource.org for
```
Comment (serhiy-storchaka) [inline LICENSE]: It is already https in Doc/license.rst.

**Review by ezio-melotti — APPROVED** [review]

File: Doc/faq/library.rst (hunk 1 above)
Comment (simon-connektica) [reply Doc/faq/library.rst:183]: Looks like the `https` link works now

Comment (serhiy-storchaka) [reply Doc/faq/library.rst:183]: Confirm. Do you mind to create a pull request?

Comment (simon-connektica) [reply Doc/faq/library.rst:183]: Looks like it has already been fixed in a different PR https://github.com/python/cpython/commit/dd53b79de0ea98af6a11481217a961daef4e9774

My mistake!

**Pull request merged**